src/main.cpp

```diff
@@ -100,26 +98,27 @@ bool isMk1Connected() {
 const string getKkInstanceName(MediaTrack* track, bool stripPrefix) {
 	int fxCount = TrackFX_GetCount(track);
 	for (int fx = 0; fx < fxCount; ++fx) {
-		// Find the Komplete Kontrol FX.
-		char rawFxName[sizeof(KK_VST3_PREFIX)];
-		TrackFX_GetFXName(track, fx, rawFxName, sizeof(rawFxName));
-		const string_view fxName(rawFxName, sizeof(rawFxName));
-		if (!fxName.starts_with(KK_VST_PREFIX) &&
-				!fxName.starts_with(KK_VST3_PREFIX)) {
-			continue;
-		}
-		// Check for the instance name.
-		// The first parameter should have a name in the form NIKBxx, where xx is a number.
-		char paramName[7];
-		TrackFX_GetParamName(track, fx, 0, paramName, sizeof(paramName));
-		const size_t prefixLen = sizeof(KK_INSTANCE_PARAM_PREFIX) - 1;
-		if (strncmp(paramName, KK_INSTANCE_PARAM_PREFIX, prefixLen) != 0) {
-			return "";
-		}
-		if (stripPrefix) {
-			return string(paramName).substr(prefixLen);
+		// Check for a matching instance name. For Komplete Kontrol, this will be the
+		// first parameter. For Kontakt, Maschine, etc., it is the last real
+		// parameter. However, REAPER appends a heap of MIDI control parameters, as
+		// well as the bypass, delta and wet parameters. Therefore, use a set of
+		// known parameter indices which will likely need to be updated in future.
+		for (int param : {0, 128, 2048, 4096}) {
+			// The focus follow ID always begins with the prefix NIXX (where XX are any
+			// alphabetical letters), and follows with a multiple digit instance number.
```

Comment:
@timrae The documentation says the number is 2 or 3 digits, but I see 4 digits with Kontakt sometimes; e.g. NIKT0018. I'm guessing this is an error in the documentation?